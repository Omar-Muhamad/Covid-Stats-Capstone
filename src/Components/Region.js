import Header from './Header';

const Region = () => (
  <>
    <Header page="region" />
    <section className="hero">
      <div className="allCountries">
        <h2>Global Covid 19 Cases</h2>
      </div>
    </section>
  </>
);

export default Region;
