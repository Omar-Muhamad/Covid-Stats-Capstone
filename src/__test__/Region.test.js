import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/configreStore';
import Region from '../Components/Region';

describe('Test Region render', () => {
  test('Test Region Render', () => {
    const tree = render(
      <Provider store={store}>
        <Region />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
