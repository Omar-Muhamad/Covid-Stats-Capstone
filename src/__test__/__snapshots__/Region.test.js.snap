// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Region component Test render Region component and mocking the store to test Regions rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header px-4 bg-[#4369B2]"
      >
        <nav
          class="nav h-[50px] text-white flex justify-between items-center font-medium "
        >
          <a
            class="flex gap-2 items-center"
            href="/"
          >
            <i
              class="fas fa-chevron-left"
            />
            <h2>
              Global
            </h2>
          </a>
          <h2>
             Stats
          </h2>
          <div
            class="flex gap-4"
          >
            <i
              class="fas fa-microphone"
            />
            <i
              class="fas fa-cog"
            />
          </div>
        </nav>
      </header>
      <section
        class="hero px-8 bg-[#5787E5] text-white h-[15rem] flex justify-center items-center gap-8"
      >
        <i
          class="fas fa-map-marked-alt fa-6x text-[#2D4573]"
        />
        <div
          class="CountryStats"
        >
          <h2
            class="text-4xl font-bold text-center"
          />
          <p
            class="text-lg text-center"
          >
            2645825 Cases
          </p>
        </div>
      </section>
      <section
        class="countriesStats"
      >
        <h2
          class="bg-[#35548B] py-[4px] px-4 text-white font-medium"
        >
          Stats by region
        </h2>
        <ul
          class="grid grid-cols-2 justify-items-end text-right bg-[#4972be]"
        >
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Tasmania
            </h2>
            <p>
              31967 Cases
            </p>
          </li>
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Victoria
            </h2>
            <p>
              889710 Cases
            </p>
          </li>
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Queensland
            </h2>
            <p>
              896530 Cases
            </p>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <header
      class="header px-4 bg-[#4369B2]"
    >
      <nav
        class="nav h-[50px] text-white flex justify-between items-center font-medium "
      >
        <a
          class="flex gap-2 items-center"
          href="/"
        >
          <i
            class="fas fa-chevron-left"
          />
          <h2>
            Global
          </h2>
        </a>
        <h2>
           Stats
        </h2>
        <div
          class="flex gap-4"
        >
          <i
            class="fas fa-microphone"
          />
          <i
            class="fas fa-cog"
          />
        </div>
      </nav>
    </header>
    <section
      class="hero px-8 bg-[#5787E5] text-white h-[15rem] flex justify-center items-center gap-8"
    >
      <i
        class="fas fa-map-marked-alt fa-6x text-[#2D4573]"
      />
      <div
        class="CountryStats"
      >
        <h2
          class="text-4xl font-bold text-center"
        />
        <p
          class="text-lg text-center"
        >
          2645825 Cases
        </p>
      </div>
    </section>
    <section
      class="countriesStats"
    >
      <h2
        class="bg-[#35548B] py-[4px] px-4 text-white font-medium"
      >
        Stats by region
      </h2>
      <ul
        class="grid grid-cols-2 justify-items-end text-right bg-[#4972be]"
      >
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Tasmania
          </h2>
          <p>
            31967 Cases
          </p>
        </li>
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Victoria
          </h2>
          <p>
            889710 Cases
          </p>
        </li>
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Queensland
          </h2>
          <p>
            896530 Cases
          </p>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Region component Test render Region component and mocking the store to test countries rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header px-4 bg-[#4369B2]"
      >
        <nav
          class="nav h-[50px] text-white flex justify-between items-center font-medium "
        >
          <a
            class="flex gap-2 items-center"
            href="/"
          >
            <i
              class="fas fa-chevron-left"
            />
            <h2>
              Global
            </h2>
          </a>
          <h2>
             Stats
          </h2>
          <div
            class="flex gap-4"
          >
            <i
              class="fas fa-microphone"
            />
            <i
              class="fas fa-cog"
            />
          </div>
        </nav>
      </header>
      <section
        class="hero px-8 bg-[#5787E5] text-white h-[15rem] flex justify-center items-center gap-8"
      >
        <i
          class="fas fa-map-marked-alt fa-6x text-[#2D4573]"
        />
        <div
          class="CountryStats"
        >
          <h2
            class="text-4xl font-bold text-center"
          />
          <p
            class="text-lg text-center"
          >
            2645825 Cases
          </p>
        </div>
      </section>
      <section
        class="countriesStats"
      >
        <h2
          class="bg-[#35548B] py-[4px] px-4 text-white font-medium"
        >
          Stats by region
        </h2>
        <ul
          class="grid grid-cols-2 justify-items-end text-right bg-[#4972be]"
        >
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Tasmania
            </h2>
            <p>
              31967 Cases
            </p>
          </li>
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Victoria
            </h2>
            <p>
              889710 Cases
            </p>
          </li>
          <li
            class="country w-full grid p-4 text-white"
            data-testid="regionListItem"
          >
            <i
              class="far fa-arrow-alt-circle-right fa-lg"
            />
            <i
              class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
            />
            <h2
              class=" text-xl font-bold"
            >
              Queensland
            </h2>
            <p>
              896530 Cases
            </p>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <header
      class="header px-4 bg-[#4369B2]"
    >
      <nav
        class="nav h-[50px] text-white flex justify-between items-center font-medium "
      >
        <a
          class="flex gap-2 items-center"
          href="/"
        >
          <i
            class="fas fa-chevron-left"
          />
          <h2>
            Global
          </h2>
        </a>
        <h2>
           Stats
        </h2>
        <div
          class="flex gap-4"
        >
          <i
            class="fas fa-microphone"
          />
          <i
            class="fas fa-cog"
          />
        </div>
      </nav>
    </header>
    <section
      class="hero px-8 bg-[#5787E5] text-white h-[15rem] flex justify-center items-center gap-8"
    >
      <i
        class="fas fa-map-marked-alt fa-6x text-[#2D4573]"
      />
      <div
        class="CountryStats"
      >
        <h2
          class="text-4xl font-bold text-center"
        />
        <p
          class="text-lg text-center"
        >
          2645825 Cases
        </p>
      </div>
    </section>
    <section
      class="countriesStats"
    >
      <h2
        class="bg-[#35548B] py-[4px] px-4 text-white font-medium"
      >
        Stats by region
      </h2>
      <ul
        class="grid grid-cols-2 justify-items-end text-right bg-[#4972be]"
      >
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Tasmania
          </h2>
          <p>
            31967 Cases
          </p>
        </li>
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Victoria
          </h2>
          <p>
            889710 Cases
          </p>
        </li>
        <li
          class="country w-full grid p-4 text-white"
          data-testid="regionListItem"
        >
          <i
            class="far fa-arrow-alt-circle-right fa-lg"
          />
          <i
            class="fas fa-map-marker-alt fa-5x text-[#2D4573] justify-self-center"
          />
          <h2
            class=" text-xl font-bold"
          >
            Queensland
          </h2>
          <p>
            896530 Cases
          </p>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
